shapeshift/web
# feat: multi hop trade ui architechtural cleanup

## Description

Rewrites most of the multi-hop trade UI to be more rebust and more react-y:

- introduces a new enum `HopExecutionState` to simplify and deduplicate orchestration of UI state based on hop execution
- refactors `MultiHopExecutionState` to be an unordered string enum so it's human readable during debugging and less prone to being wrong with refactors
- centralized all incrementation of `tradeExecutionState` into the trade quote slice, with all logic centralized in a function `getNextTradeExecutionState`
- added initial fetch of `isApprovalNeeded` per-hop to allow loading state and correct rendering of UI steps initially, rather than the view layer managing this
- combined `FirstHop` and `SecondHop` into single `Hop` component with complete re-use of logic and hooks
- split up the functions in `steps.tsx` into ackchyal react components and utils
- general renaming of `*ExecutionStatus` to `*ExecutionState` since those values are variables in a state machine, not a  status machine
- Added a clause to `checkApprovalNeeded` to ignore non-evm sell assets 

## Pull Request Type


## Issue (if applicable)

NA

## Risk

- Modifies non-prod facing parts of the tradeQuoteSlice, so may pose low risk of broken trades in production
- Adds a clause to `checkApprovalNeeded` to ignore non-evm sell assets, which may break approval checks somehow

## Testing

- check trades are still working
- check approvals are still working for erc20s+ bep20s
- check non-evm trades dont want erc20/bep20s approvals

### Engineering



### Operations



## Screenshots (if applicable)
